qiskit-qir/src/qiskit_qir/elements.py

```diff
@@ -0,0 +1,102 @@
+##
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+##
+from dis import Instruction
+from typing import List, Union
+from cirq import num_qubits
+from qiskit import ClassicalRegister, QuantumRegister
+from qiskit.circuit.bit import Bit
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+
+
+class _QuantumCircuitElement:    
+    @classmethod
+    def from_element_list(cls, elements):
+        return [cls(elem) for elem in elements]
+
+
+class _Register(_QuantumCircuitElement):    
+    def __init__(self, register: Union[QuantumRegister, ClassicalRegister]):
+        self._register = register
+
+    def accept(self, visitor):
+        visitor.visit_register(self._register)
+
+
+class _Instruction(_QuantumCircuitElement):
+    def __init__(
+        self,
+        instruction: Instruction,
+        qargs: List[Bit],
+        cargs: List[Bit]
+    ):
+        self._instruction = instruction
+        self._qargs = qargs
+        self._cargs = cargs
+
+    def accept(self, visitor):
+        visitor.visit_instruction(self._instruction, self._qargs, self._cargs)
+
+
+class QiskitModule:
```

Comment:
I'm not quite sure I understand the name "module" here; it looks from usage below like this this is effectively a wrapper that allows visiting parts of a circuit in order to produce modules, rather than a module itself?